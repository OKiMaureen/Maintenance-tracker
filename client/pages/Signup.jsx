import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import signUpAction from '../actions/signUpAction';
import loader from '../assets/images/spinloader.gif';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class Signup extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      name: '',
      email: '',
      password: '',
      password_confirmation: '',
      errors: {},
    }

    validate = () => {
      const validations = new Validator(
        {
          name: this.state.name,
          email: this.state.email,
          password: this.state.password,
          password_confirmation: this.state.password_confirmation,
        },
        {
          name: ['required', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:15', 'string'],
          email: 'required|email|string',
          password: 'required|min:8|max:30|string',
          password_confirmation: 'required',
        },
        {
          'min.password': 'The :attribute must not be less than 9 characters.',
          'max.password': 'The :attribute must not be greater than 30 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.state.password !== this.state.password_confirmation) {
        this.setState({
          errors: {
            password_confirmation: ['Password and password confirmation must be the same'],
          },
        });
      }

      if (this.validate()) {
        this.props.signUpAction(this.state, this.props.history);
      }
      return true;
    }

    render() {
      return (
        <div>
          {/* Signup Form  */}
          <Header><Link to="/">Home</Link><Link to="/" /></Header>
          <section className="form">
            <article className="card">
              <form id="signupForm">
                <h3>Sign Up</h3>
                <br />
                {this.props.userDetail.checkStatus.isLoading ? (<span className="loader"><img src={loader} alt="loader" /></span>) : '' }
                <input type="text" id="name" name="name" placeholder="Name" onChange={this.handleChange} />
                {
                  this.state.errors.name ? this.state.errors.name.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
                <br />
                <input type="text" id="email" name="email" placeholder="Email" onChange={this.handleChange} required />
                {
                  this.state.errors.email ? this.state.errors.email.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }{
                  this.props.userDetail.error ? <span className="validation-error">{this.props.userDetail.error}</span> : ''
                }

                <br />
                <input type="password" id="password" name="password" placeholder="Password" onChange={this.handleChange} required />
                {
                  this.state.errors.password ? this.state.errors.password.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
                <br />
                <input type="password" id="confirmPassword" name="password_confirmation" placeholder="Confirm Password" onChange={this.handleChange} required />
                {
                  this.state.errors.password_confirmation ? this.state.errors.password_confirmation.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
                <br />
                <button type="submit" className="button" id="signupBtn" onClick={this.handleSubmit} >Sign Up</button>
                <p>Already have an account?
                  <Link to="/signin">Sign In</Link>
                </p>
              </form>
            </article>
          </section>
          {/* End signup form */}
          <Footer />
        </div>

      );
    }
}

const mapStateToProps = state => ({
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  signUpAction,
}, dispatch);

Signup.propTypes = {
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  signUpAction: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(Signup);

