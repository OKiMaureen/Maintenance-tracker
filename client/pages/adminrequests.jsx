import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Header from '../components/Header';
import adminRequestAction from '../actions/adminRequestAction';
import AllRequestsComponent from '../components/AllRequests';
import logoutAction from '../actions/logoutAction';

export class AdminRequests extends Component {
  componentDidMount() {
    const { adminRequests } = this.props;
    adminRequests();
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  render() {
    const { requests = [] } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="admincreaterequest">Create Request</Link>
          <Link className="current" to="adminrequest">All Request</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <AllRequestsComponent
          requests={requests}
          logout={this.logout}
          linkRoute="/adminrequests"
          linkText="All Requests"
          route="/requestdetails/"
        />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  requests: state.adminRequests.adminRequests,
  error: state.adminRequests.error,
  userDetail: state.authUser,

});


const mapDispatchToProps = dispatch => bindActionCreators({
  adminRequests: adminRequestAction,
  logout: logoutAction,
}, dispatch);

AdminRequests.propTypes = {
  requests: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  adminRequests: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AdminRequests);
