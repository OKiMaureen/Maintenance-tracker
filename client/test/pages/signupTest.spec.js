import React from 'react';
import { expect as e } from 'chai';
import { shallow } from 'enzyme';
import { Signup } from '../../pages/Signup';


describe('<SignUp />', () => {
  let wrapper;
  const props = {
    signUpAction: () => {},
    userDetail: {
      authUser: {
        id: 1,
        name: 'maureen',
        email: 'ccureen.eloho@gmail.com',
        role: 'user',
      },
      checkStatus: {
        isLoading: false,
        success: false,
        error: false,
      },
    },
    history: {},
    error: '',
  };


  it('Should return number of input field on Signup page', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.find('input')).to.have.length(4);
  });
  it('stores a snapshot of the component', () => {
    wrapper = shallow(<Signup {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should return number of button on Signup page', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.find('button')).to.have.length(1);
  });
  it('Should check if the wrapper contains instance of Signup page', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance()).to.be.instanceof(Signup);
  });

  it('Should check if name is supplied on successful signup', () => {
    wrapper = shallow(<Signup {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, name: 'maureen' });
    e(wrapper.state().name).to.be.equal('maureen');
  });

  it('the validator function should return true when validation is successfull', () => {
    wrapper = shallow(<Signup {...props} />);
    wrapper.setState({
      name: 'maureen',
      email: 'maureen@mail.ru',
      password: 'password',
      password_confirmation: 'password',
    });
    const { validate } = wrapper.instance();

    const result = validate();

    expect(result).toBe(true);
  });
  it('should check that the handlechange function should clear the errors  when valid input is written', () => {
    wrapper = shallow(<Signup {...props} />);
    wrapper.setState({
      name: 'maureen',
      email: 'maureen@mail.com',
      password: 'password',
      password_confirmation: 'password',
      errors: {
        name: ['the name is required.'],
      },
    });
    const { handleChange } = wrapper.instance();
    handleChange({
      target: {
        value: 'maureen',
        name: 'name',
      },
    });

    expect(wrapper.state().errors.name).toBe('');
  });

  it('the handlesubmit function should return error if password and password confirmation do not match', () => {
    wrapper = shallow(<Signup {...props} />);
    wrapper.setState({
      name: 'maureen',
      email: 'maureen@mail.com',
      password: 'password',
      password_confirmation: 'Notpassword',
      errors: {
        password_confirmation: ['Password and password confirmation must be the same'],
      },
    });
    const { handleSubmit } = wrapper.instance();
    handleSubmit({ preventDefault: () => {} });

    expect(wrapper.state().errors.password_confirmation).toEqual(['Password and password confirmation must be the same']);
  });

  it('Should check if email is supplied on successful signup', () => {
    wrapper = shallow(<Signup {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, email: 'maureen@gmail.com' });
    e(wrapper.state().email).to.be.equal('maureen@gmail.com');
  });
  it('Should check if password is supplied on successful signup', () => {
    wrapper = shallow(<Signup {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, password: 'maureen' });
    e(wrapper.state().password).to.be.equal('maureen');
  });
  it('Should check if password_confirmation is supplied on the Signup page', () => {
    wrapper = shallow(<Signup {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, password_confirmation: 'maureen' });
    e(wrapper.state().password_confirmation).to.be.equal('maureen');
  });
  it('Should check if password_confirmation is supplied on the Signup page', () => {
    wrapper = shallow(<Signup {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, errors: '{}' });
    e(wrapper.state().errors).to.be.equal('{}');
  });

  it('should set user name to state when user types', () => {
    wrapper = shallow(<Signup {...props} />);
    const input = wrapper.find('input[name="name"]');
    input.simulate('change', {
      target: {
        value: 'maureen',
        name: 'name',
      },
    });

    e(wrapper.state().name).to.equal('maureen');
  });

  it('Should handle and store user details', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'name' } })).to.be.equal(true);
  });

  it('Should handle and store user details', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'email' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'password' } })).to.be.equal(true);
  });

  it('Should handle user details validation', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance().validate()).to.be.equal(false);
  });

  it('Should handle signing up new user', () => {
    wrapper = shallow(<Signup {...props} />);
    e(wrapper.instance().handleSubmit({ preventDefault: () => {} })).to.be.equal(true);
  });
  it('Should handle loading a loader', () => {
    wrapper = shallow(<Signup
      {...props}
      userDetail={{
      checkStatus: {
        isLoading: true,
      },
    }}
    />);
    expect(wrapper.find('.loader').length).toBe(1);
  });
  it('Should handle loading a not loading on success or error', () => {
    wrapper = shallow(<Signup
      {...props}
      userDetail={{
      checkStatus: {
        isLoading: false,
      },
    }}
    />);
    expect(wrapper.find('.loader').length).toBe(0);
  });
  it('Should handle loading the error message', () => {
    wrapper = shallow(<Signup
      {...props}
      userDetail={{
        checkStatus: {
          isLoading: false,
      },

      error: 'Email already exists',
    }}
    />);
    expect(wrapper.find('.validation-error').length).toBe(1);
  });
  it('Should handle  not loading the error message on success or loading', () => {
    wrapper = shallow(<Signup
      {...props}
      userDetail={{
        checkStatus: {
          isLoading: false,
      },

      error: '',
    }}
    />);
    expect(wrapper.find('.validation-error').length).toBe(0);
  });
});
