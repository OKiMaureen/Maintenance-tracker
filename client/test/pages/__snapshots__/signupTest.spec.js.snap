// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> stores a snapshot of the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signup
    error=""
    history={Object {}}
    signUpAction={[Function]}
    userDetail={
      Object {
        "authUser": Object {
          "email": "ccureen.eloho@gmail.com",
          "id": 1,
          "name": "maureen",
          "role": "user",
        },
        "checkStatus": Object {
          "error": false,
          "isLoading": false,
          "success": false,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
          <Link
            replace={false}
            to="/signin"
          >
            Sign In
          </Link>
        </Header>,
        <section
          className="form"
        >
          <article
            className="card"
          >
            <form
              id="signupForm"
            >
              <h3>
                Sign Up
              </h3>
              <br />
              
              <input
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              />
              
              <br />
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <input
                id="confirmPassword"
                name="password_confirmation"
                onChange={[Function]}
                placeholder="Confirm Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign Up
              </button>
              <p>
                Already have an account?
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign In
                </Link>
              </p>
            </form>
          </article>
        </section>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
            <Link
              replace={false}
              to="/signin"
            >
              Sign In
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sign In",
              "replace": false,
              "to": "/signin",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <article
            className="card"
          >
            <form
              id="signupForm"
            >
              <h3>
                Sign Up
              </h3>
              <br />
              
              <input
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              />
              
              <br />
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <input
                id="confirmPassword"
                name="password_confirmation"
                onChange={[Function]}
                placeholder="Confirm Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign Up
              </button>
              <p>
                Already have an account?
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign In
                </Link>
              </p>
            </form>
          </article>,
          "className": "form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              id="signupForm"
            >
              <h3>
                Sign Up
              </h3>
              <br />
              
              <input
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Name"
                type="text"
              />
              
              <br />
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <input
                id="confirmPassword"
                name="password_confirmation"
                onChange={[Function]}
                placeholder="Confirm Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign Up
              </button>
              <p>
                Already have an account?
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign In
                </Link>
              </p>
            </form>,
            "className": "card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Sign Up
                </h3>,
                <br />,
                "",
                <input
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                />,
                "",
                <br />,
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                />,
                "",
                "",
                <br />,
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />,
                "",
                <br />,
                <input
                  id="confirmPassword"
                  name="password_confirmation"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={true}
                  type="password"
                />,
                "",
                <br />,
                <button
                  className="button"
                  id="signupBtn"
                  onClick={[Function]}
                  type="submit"
                >
                  Sign Up
                </button>,
                <p>
                  Already have an account?
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </p>,
              ],
              "id": "signupForm",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "name",
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Name",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "required": true,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "required": true,
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "confirmPassword",
                  "name": "password_confirmation",
                  "onChange": [Function],
                  "placeholder": "Confirm Password",
                  "required": true,
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                  "className": "button",
                  "id": "signupBtn",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Already have an account?",
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Already have an account?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign In",
                      "replace": false,
                      "to": "/signin",
                    },
                    "ref": null,
                    "rendered": "Sign In",
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
            ],
            "type": "form",
          },
          "type": "article",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              replace={false}
              to="/signin"
            >
              Sign In
            </Link>
          </Header>,
          <section
            className="form"
          >
            <article
              className="card"
            >
              <form
                id="signupForm"
              >
                <h3>
                  Sign Up
                </h3>
                <br />
                
                <input
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                />
                
                <br />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                />
                
                
                <br />
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <input
                  id="confirmPassword"
                  name="password_confirmation"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <button
                  className="button"
                  id="signupBtn"
                  onClick={[Function]}
                  type="submit"
                >
                  Sign Up
                </button>
                <p>
                  Already have an account?
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </p>
              </form>
            </article>
          </section>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
              <Link
                replace={false}
                to="/signin"
              >
                Sign In
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign In",
                "replace": false,
                "to": "/signin",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <article
              className="card"
            >
              <form
                id="signupForm"
              >
                <h3>
                  Sign Up
                </h3>
                <br />
                
                <input
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                />
                
                <br />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                />
                
                
                <br />
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <input
                  id="confirmPassword"
                  name="password_confirmation"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <button
                  className="button"
                  id="signupBtn"
                  onClick={[Function]}
                  type="submit"
                >
                  Sign Up
                </button>
                <p>
                  Already have an account?
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </p>
              </form>
            </article>,
            "className": "form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                id="signupForm"
              >
                <h3>
                  Sign Up
                </h3>
                <br />
                
                <input
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  type="text"
                />
                
                <br />
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                />
                
                
                <br />
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <input
                  id="confirmPassword"
                  name="password_confirmation"
                  onChange={[Function]}
                  placeholder="Confirm Password"
                  required={true}
                  type="password"
                />
                
                <br />
                <button
                  className="button"
                  id="signupBtn"
                  onClick={[Function]}
                  type="submit"
                >
                  Sign Up
                </button>
                <p>
                  Already have an account?
                  <Link
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </p>
              </form>,
              "className": "card",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Sign Up
                  </h3>,
                  <br />,
                  "",
                  <input
                    id="name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                  />,
                  "",
                  <br />,
                  <input
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                  />,
                  "",
                  "",
                  <br />,
                  <input
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />,
                  "",
                  <br />,
                  <input
                    id="confirmPassword"
                    name="password_confirmation"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    required={true}
                    type="password"
                  />,
                  "",
                  <br />,
                  <button
                    className="button"
                    id="signupBtn"
                    onClick={[Function]}
                    type="submit"
                  >
                    Sign Up
                  </button>,
                  <p>
                    Already have an account?
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>
                  </p>,
                ],
                "id": "signupForm",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "name",
                    "name": "name",
                    "onChange": [Function],
                    "placeholder": "Name",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "email",
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "Email",
                    "required": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "",
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "password",
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "Password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "confirmPassword",
                    "name": "password_confirmation",
                    "onChange": [Function],
                    "placeholder": "Confirm Password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                    "className": "button",
                    "id": "signupBtn",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Already have an account?",
                      <Link
                        replace={false}
                        to="/signin"
                      >
                        Sign In
                      </Link>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Already have an account?",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Sign In",
                        "replace": false,
                        "to": "/signin",
                      },
                      "ref": null,
                      "rendered": "Sign In",
                      "type": [Function],
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "form",
            },
            "type": "article",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
